hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java

```diff
@@ -79,6 +84,37 @@ public void enrichEvent(StructuredTrace trace, Event event) {
     // does not need to build the full traversal graph, just get the parents mapping
     StructuredTraceGraph graph = buildGraph(trace);
 
+    if (null == graph.getTraceEntitiesGraph() || null == graph.getTraceEventsGraph()) {
```

Comment:
This can happen from multiple places, and we have seen in other enrichers too. So, should we move this to the data-model class?

Reply:
Moved the log, on the other suggestion should we first observe the log and then make changes?